Don't automagically add "[OPTIONS]" to usage

thaJeztah opened the pull request:
This removes the logic to automatically add [OPTIONS] to the usage output.

The current logic was broken if a command only has deprecated or hidden flags, and in many cases put the [OPTIONS] in the wrong location.

Requiring the usage string to be set manually gives more predictable results, and shouldn't require much to maintain.

thaJeztah committed:
Don't automagically add "[OPTIONS]" to usage

This removes the logic to automatically
add [OPTIONS] to the usage output.

The current logic was broken if a command
only has deprecated or hidden flags, and
in many cases put the [OPTIONS] in the
wrong location.

Requiring the usage string to be set
manually gives more predictable results,
and shouldn't require much to maintain.

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah commented:
/cc @dnephin @vdemeester ptal :smile:

dnephin commented:
LGTM, but I've been wondering if we should even just remove this condition from the template entirely. I think in almost every case we would include `[OPTIONS]` in the use line, because it often needs to appear before the positional arguments.

It's also more obvious if the use line is just printed directly instead of having this added to it.

WDYT?

thaJeztah commented:
Yes, I agree with that; takes a lot of the guesswork out of it. I think there was another location that also added the `[OPTIONS]`. I'll have a look and update this PR

thaJeztah commented:
ping @dnephin updated; I removed all logic to automatically add `[OPTIONS]` to the usage output, and updated commands that did not yet include `[OPTIONS]` in the usage / `use:` string.

vdemeester commented:
LGTM 👼

dnephin commented:
LGTM
